### ISSUE TITLE
Customizable Rollups Exception for Filter Rule with Contains Operation for Picklist

### ISSUE THREAD

**Issue by jesslopez-sf** [SalesforceFoundation/Cumulus#3675]
When a Filter Rule uses a Contains operator for a picklist field, the Customizable Rollups batch job fails with a null pointer exception. The Queueable works fine initially in my scratch org when creating the Opportunity and populates the rollup field. When clearing out the field, the Recalculate Rollups button works as expected in my scratch org, but I do have a Case where this is not working either.  The batch job shows the Exception. When I switch the operation to “equals”, it works without error. If I use a Contains filter on a regular text field (vs. a picklist, for example on the Opportunity Name), it works fine.

![operator](https://user-images.githubusercontent.com/29731456/46413381-cb4ae780-c6ee-11e8-973d-98f6c41ee793.jpg)

Other operators in the Filter Rule list for single and multi-picklist fields are:

* Equals
* Not Equals
* Starts With
* Contains
* Does Not Contain
* In List
* Not In List

One more thing to note: I replicated this using a standard picklist field on the Opportunity. In the customer org I was troubleshooting, they had a single picklist on Campaign (Single_Picklist__c). They then had a text formula on the Opportunity (Single_Picklist__c) which had a formula of TEXT(Campaign.Single_Picklist__c). The Filter Rule then used a contains on the formula field (Opportunity: Single Picklist contains 'abc123').

Debug logs will be attached to the Work Item.

### Stack Trace ### 
Stacktrace available at the time of saving the record - 
```
Class.CMT_FilterRule.CompareStrings.isIncluded: line 412, column 1
Class.CMT_FilterRule.isIncluded: line 162, column 1
Class.CMT_FilterRuleEvaluation_SVC.evaluateFilterGroup: line 115, column 1
Class.CRLP_RollupAccount_SVC.processRollups: line 80, column 1
Class.CRLP_RollupProcessor_SVC.completeRollupForParent: line 303, column 1
Class.CRLP_RollupProcessor_SVC.startRollupProcess: line 115, column 1
Class.CRLP_Batch_Base_NonSkew.execute: line 286, column 1
```

### Steps to Repeat ###

**Setup:**

1. Create Account field 
    1. Type = Currency
    2. Name = Test Amount
    3. Length: 16
    4. Decimal Places: 2
2. Enable Customizable Rollups in NPSP Settings → Donations → Customizable Rollups
3. Create a Filter Group
    1. Name = Test Filter Group
    2. Filter Rules
        1. Object: Opportunity
        2. Field: Opportunity Type
        3. Operator: Contains
        4. Value = New Customer
4. Create a Rollup
    1. Target Object: Account
    2. Target Field: Test Amount
    3. Operation: Sum
    4. Time Frame: All Time
    5. Rollup Type: Opportunity → Account (Hard Credit)
    6. Filter Group: Test Filter Group
    7. Amount Field: Opportunity: Amount

**Test 1: Works as expected**

1. Create a Contact: Jess Lopez
2. Create an Opportunity for the Jess Lopez Household (Close Date = 10/2/2018, Amount = 100, Stage = Closed Won, Opportunity Type = New Customer)
3. The Test Amount field is populated correctly on the Account ($100). 

**Test 2: Exception thrown**

1. Clear out the Test Amount field on the Lopez Household Account record.
2. Go to NPSP Settings → Bulk Data Processes → Rollup Donations Batch and run the batch.
3. An error is captured in the Errors log with the stack trace information listed above.

**Comment by jesslopez-sf** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007fOKQAA2" target="_blank">W-028850</a>

**Comment by mrbelvedere** [comment]
Included in beta release 3.142 (Beta 4)
